extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitterConfig.java

```diff
@@ -21,53 +21,140 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
 import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.apache.druid.java.util.common.StringUtils;
 import org.apache.kafka.clients.producer.ProducerConfig;
 
 import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
 
 public class KafkaEmitterConfig
 {
+  public enum EventType
+  {
+    METRICS,
+    ALERTS,
+    REQUESTS,
+    SEGMENT_METADATA{
+      @Override
+      public String toString()
+      {
+        return "segmentMetadata";
+      }
```

Comment:
we also need to override the fromString method. As it stands passing `segmentMetadata` as a config value would throw an exception when de-serializing the config because only `segment_metadata` would work and not `segmentMetadata`

let's make sure we include that in the test for event types config, since this that would have caught the problem

Reply:
Updated as we discussed